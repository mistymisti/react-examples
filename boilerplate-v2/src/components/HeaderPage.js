import React from 'react';
import { NavLink } from 'react-router-dom';
import { connect } from 'react-redux';
import { startLogOut } from '../actions/auth';

export const HeaderPage = ({ startLogOut }) => (
    <header>
        <h1>BoilderPlate React App</h1>
        <div>
            <NavLink to="/dashboard" activeClassName="" >Home</NavLink>
        </div>
        <div>
            <NavLink to="/create" activeClassName="" >Create</NavLink>
        </div>
        <div>    
            <NavLink to="/edit" activeClassName="">Edit</NavLink>
        </div>
        <button onClick= { startLogOut }>LogOut</button>
    </header>
);

const mapDispatchToProps = (dispatch) => ({
    startLogOut: () => dispatch(startLogOut())
});

export default connect(undefined, mapDispatchToProps)(HeaderPage);