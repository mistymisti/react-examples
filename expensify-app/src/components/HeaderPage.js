import React from 'react';
import { NavLink } from 'react-router-dom';

const HeaderPage = () => (
    <header>
        <h1>Expensify</h1>
        <div>
            <NavLink to="/" activeClassName="" >Home</NavLink>
        </div>
        <div>
            <NavLink to="/create" activeClassName="" >Create</NavLink>
        </div>
        <div>    
            <NavLink to="/edit" activeClassName="">Edit</NavLink>
        </div>
        <div>
            <NavLink to="/help" activeClassName="">Help</NavLink>
        </div>
    </header>
);

export default HeaderPage;