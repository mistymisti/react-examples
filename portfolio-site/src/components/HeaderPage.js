import React from 'react';
import { NavLink } from 'react-router-dom';

const HeaderPage = () => (
    <header>
        <h1>Portfolio</h1>
        <div>
            <NavLink to="/" activeClassName="" >Home</NavLink>
        </div>
        <div>
            <NavLink to="/portfolio" activeClassName="" >Portfolio</NavLink>
        </div>
        <div>    
            <NavLink to="/contact" activeClassName="">Contact</NavLink>
        </div> 
    </header>
);

export default HeaderPage;