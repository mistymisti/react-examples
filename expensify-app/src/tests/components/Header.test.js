import React from 'react';
import ReactShallowRenderer from 'react-test-renderer/shallow';
import HeaderPage from '../../components/HeaderPage';
import { shallow } from 'enzyme';

test('should test the header component', ()=> {
    const wrapper = shallow(<HeaderPage />);
    expect(wrapper.find('h1').length).toBe(1);
    expect(wrapper).toMatchSnapshot();
   // expect(wrapper.find('h1').text()).toBe('Expensify');
    // const reactShallowRender = new ReactShallowRenderer();
    // reactShallowRender.render(<HeaderPage />);
    // expect(reactShallowRender.getRenderOutput()).toMatchSnapshot();        
});