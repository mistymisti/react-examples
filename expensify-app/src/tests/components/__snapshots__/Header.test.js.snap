// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should test the header component 1`] = `
<header>
  <h1>
    Expensify
  </h1>
  <div>
    <NavLink
      activeClassName=""
      ariaCurrent="true"
      to="/"
    >
      Home
    </NavLink>
  </div>
  <div>
    <NavLink
      activeClassName=""
      ariaCurrent="true"
      to="/create"
    >
      Create
    </NavLink>
  </div>
  <div>
    <NavLink
      activeClassName=""
      ariaCurrent="true"
      to="/edit"
    >
      Edit
    </NavLink>
  </div>
  <div>
    <NavLink
      activeClassName=""
      ariaCurrent="true"
      to="/help"
    >
      Help
    </NavLink>
  </div>
</header>
`;
